test/hotspot/jtreg/testlibrary_tests/template_framework/examples/TestTutorial.java

```diff
@@ -673,17 +856,19 @@ public static String generateWithDataNamesForFieldsAndVariables() {
         ));
 
         // Define a local variable.
-        var templateLocalVariable = Template.make("type", (DataName.Type type) -> body(
-            addDataName($("var"), type, MUTABLE),
+        // Note: it is very important that we use a "transparentScope" for the template here,
+        //       so that the DataName can escape to outer scopes.
```

Comment:
Might be obvious but you could add that it does not mean it's available for the entire outer scope but only for everything that follows this definition/insertion of the template.